cargo-pgrx/src/command/upgrade.rs

```diff
@@ -0,0 +1,331 @@
+//LICENSE Portions Copyright 2019-2021 ZomboDB, LLC.
+//LICENSE
+//LICENSE Portions Copyright 2021-2023 Technology Concepts & Design, Inc.
+//LICENSE
+//LICENSE Portions Copyright 2023-2023 PgCentral Foundation, Inc. <contact@pgcentral.org>
+//LICENSE
+//LICENSE All rights reserved.
+//LICENSE
+//LICENSE Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+use cargo_edit::{registry_url, update_registry_index, Dependency, LocalManifest};
+use eyre::eyre;
+use std::path::{Path, PathBuf};
+use toml_edit::KeyMut;
+use tracing::{debug, error, info, warn};
+
+use crate::CommandExecute;
+
+/// Upgrade pgrx crate versions in `Cargo.toml`.
+/// Defaults to latest.
+#[derive(clap::Args, Debug)]
+#[clap(author)]
+pub(crate) struct Upgrade {
+    /// Specify a version to upgrade to, rather than defaulting to the latest
+    /// version.
+    #[clap(long, short)]
+    pub(crate) target_version: Option<String>, //TODO: semver not string
+
+    /// Path to the manifest file (usually Cargo.toml). Defaults to
+    /// "./Cargo.toml" in the working directory.
+    #[clap(long = "manifest-path", short)]
+    pub(crate) manifest_path: Option<PathBuf>,
+
+    /// Flag for upgrading PGRX to pre-release versions.
+    #[clap(long = "include-prereleases")]
+    pub(crate) include_prereleases: bool,
+
+    /// Select a package within the current workspace within which to upgrade
+    /// pgrx package versions. Defaults to the root manifest in the working
+    /// directory.
+    #[clap(long, short)]
+    pub(crate) package: Option<String>,
+
+    /// Dry-run - if this flag is set, Cargo.toml will not be modified.
+    /// Instead, this command will print the text of the new Cargo.toml
+    /// that would have been generated if it was modified.
+    #[clap(long = "dry-run", short)]
```

Comment:
Cargo always uses this for its "dry run" short form, idk why.
```suggestion
    #[clap(long = "dry-run", short = "n")]
```